frontend/src/metabase/query_builder/components/QueryModals.jsx

```diff
@@ -171,11 +180,21 @@ export default class QueryModals extends React.Component {
           initialCollectionId={question.collectionId()}
           onClose={onCloseModal}
           onMove={collection => {
-            const card = question
-              .setCollectionId(collection && collection.id)
-              .card();
-
-            this.props.onSave(card);
+            this.props.setQuestionCollection(
+              { id: question.id() },
+              collection,
+              {
+                notify: {
+                  message: (
+                    <QuestionMoveToast
+                      isModel={question.isDataset()}
+                      collectionId={collection.id || ROOT_COLLECTION.id}
+                    />
+                  ),
+                  undo: false,
```

Comment:
Remove undo actions because it's not working. The screenshot is from moving a dashboard, but it should look really similar.
![image](https://user-images.githubusercontent.com/1937582/177955726-68f22aa8-6d65-4e9f-8ef3-2f8da1180f45.png)